serialize feature do not compile

Trying to use the serialize feature result in a error:

```
$ cargo test --features serialize 
   Compiling proc-macro2 v1.0.95
   Compiling unicode-ident v1.0.18
   Compiling serde v1.0.219
   Compiling isbn v0.4.1 (/var/home/misc/checkout/git/isbn-rs)
   Compiling quote v1.0.40
   Compiling syn v2.0.104
   Compiling serde_derive v1.0.219
error[E0277]: the trait bound `&'a Isbn10: Deserialize<'_>` is not satisfied
    --> src/lib.rs:175:9
     |
175  |     _10(&'a Isbn10),
     |         ^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `&'a Isbn10`
     |
     = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `&'a Isbn10` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
note: required by a bound in `newtype_variant`
    --> /home/misc/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/serde-1.0.219/src/de/mod.rs:2125:12
     |
2123 |     fn newtype_variant<T>(self) -> Result<T, Self::Error>
     |        --------------- required by a bound in this associated function
2124 |     where
2125 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `VariantAccess::newtype_variant`
help: consider removing the leading `&`-reference
     |
175  -     _10(&'a Isbn10),
175  +     _10(Isbn10),
     |

error[E0277]: the trait bound `&'a Isbn13: Deserialize<'_>` is not satisfied
    --> src/lib.rs:176:9
     |
176  |     _13(&'a Isbn13),
     |         ^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `&'a Isbn13`
     |
     = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `&'a Isbn13` type
     = note: for types from other crates check whether the crate offers a `serde` feature flag
note: required by a bound in `newtype_variant`
    --> /home/misc/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/serde-1.0.219/src/de/mod.rs:2125:12
     |
2123 |     fn newtype_variant<T>(self) -> Result<T, Self::Error>
     |        --------------- required by a bound in this associated function
2124 |     where
2125 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `VariantAccess::newtype_variant`
help: consider removing the leading `&`-reference
     |
176  -     _13(&'a Isbn13),
176  +     _13(Isbn13),
     |

For more information about this error, try `rustc --explain E0277`.
error: could not compile `isbn` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: could not compile `isbn` (lib test) due to 2 previous errors
```

My toolchain:
```
$ cargo --version ; rustc --version
cargo 1.88.0 (873a06493 2025-05-10) (Fedora 1.88.0-1.fc42)
rustc 1.88.0 (6b00bc388 2025-06-23) (Fedora 1.88.0-1.fc42)
```

Removing the derive for the IsbnRef struct fix the error:
```patch
diff --git a/src/lib.rs b/src/lib.rs
index 38fd8a4..90678b7 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -170,7 +170,6 @@ impl Isbn {
 /// assert_eq!("978-1-4920-6766-5".parse(), Ok(isbn_13));
 /// ```
 #[derive(Debug, PartialEq, Clone, Eq)]
-#[cfg_attr(feature = "serialize", derive(Serialize, Deserialize))]
 pub enum IsbnRef<'a> {
     _10(&'a Isbn10),
     _13(&'a Isbn13),
```

But this is not really the right fix, obviously.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -170,7 +170,6 @@ impl Isbn {
 /// assert_eq!("978-1-4920-6766-5".parse(), Ok(isbn_13));
 /// ```
 #[derive(Debug, PartialEq, Clone, Eq)]
-#[cfg_attr(feature = "serialize", derive(Serialize, Deserialize))]
 pub enum IsbnRef<'a> {
     _10(&'a Isbn10),
     _13(&'a Isbn13),

